sw/device/silicon_creator/lib/drivers/hmac_functest.c

```diff
@@ -0,0 +1,67 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/runtime/print.h"
+#include "sw/device/silicon_creator/lib/drivers/hmac.h"
+#include "sw/device/silicon_creator/lib/error.h"
+#include "sw/device/silicon_creator/lib/test_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+static const char kGettysburgPrelude[] =
+    "Four score and seven years ago our fathers brought forth on this "
+    "continent, a new nation, conceived in Liberty, and dedicated to the "
+    "proposition that all men are created equal.";
+
+// The following shell command will produce the sha256sum, although the words
+// will be in the reverse order that they appear in the digest array
+// returned by `hmac_sha256_final`.
+//
+// $ echo -n "Four score and seven years ago our fathers brought forth on this
+// continent, a new nation, conceived in Liberty, and dedicated to the
+// proposition that all men are created equal." |
+//     sha256sum - | cut -f1 -d' ' | sed -e "s/......../0x&, /g"
+//
+// Since the shell command above emits the words in reversed order, we store
+// them here in reversed order and perform the test with that in mind.
+//
+// TODO(lowrisc/opentitan#6644): This should be static, but extract_sw_logs.py
+// complains about duplicate strings when this is marked static.
```

Comment:
This is probably because `extract_sw_logs.py` doesn't handle the case when `str.find()` returns -1. I ran into the same issue when cross-compiling for coverage. Created #6647.

Reply:
#6647 is now merged.